collector/__init__.py

```diff
@@ -1,504 +1,507 @@
-                    instrument_name = instrument_config['name'][0]
-
-                # ToO status
-                too_status = get_obs_too_status(raw_observation, self.programs[site][program_id].too_status)
-
-                # Sky Conditions
-                conditions = get_conditions(raw_observation, label=False)
-
-                if conditions is None or not conditions:
-                    sky_cond = SkyConditions()
-                else:
-                    parse_conditions = conditions_parser(conditions)                    
-                    sky_cond = SkyConditions(*parse_conditions)
-
-                # Elevation constraints        
-                elevation_type, min_elevation, max_elevation = get_elevation(raw_observation)
-                elevation_type = str_to_elevation_type(elevation_type)
-                min_elevation, max_elevation = str_to_float(min_elevation), str_to_float(max_elevation)
-                elevation_constraints = ElevationConstraints(elevation_type, min_elevation, max_elevation)
-
-                # Charged observation time
-                # Used time from Time Accounting Summary (tas) information
-                obs_time = 0
-                if tas and program_id in tas and obs_odb_id in tas[program_id]:
-                    obs_time = max(0, tas[program_id][obs_odb_id]['prgtime'].value)
-
-                # Total observation time
-                # for IGRINS, update tot_time to take telluric into account if there is time remaining
-                calibration_time = 0
-                if 'IGRINS' in instrument_name.upper() and total_time.total_seconds() / 3600. - obs_time > 0.0:
-                    calibration_time = 10 / 60  # fractional hour
-
-                inst_config = Collector._instrument_setup(instrument_config, instrument_name)
-
-                start, duration, repeat, period = get_windows(raw_observation)
-                # This makes a list of timing windows between progstart and progend
-                timing_windows = ot_timing_windows(start, duration, repeat, period)
-
-                progstart = collected_program.start
-                progend = collected_program.end
-
-                windows = []
-                if len(timing_windows) == 0:
-                    windows.append(Time([progstart, progend]))
+            windows = []
+            for tw in obs.constraints.timing_windows:
+                t0 = time.mktime(tw.start.utctimetuple()) * 1000 * u.ms
+                begin = Time(t0.to_value('s'), format='unix', scale='utc')
+                duration = tw.duration / 3600000.0 * u.h
+                repeat = max(1, tw.repeat)
+                period = tw.period.total_seconds() / 3600.0 * u.h if tw.period is not None else 0.0 * u.h
+                windows.extend([Time([begin + i * period, begin + i * period + duration]) for i in range(repeat)])
+
+        self._timing_windows[obs.id] = windows
+
+    def _calculate_target_info(self,
```

Comment:
Yes, very possibly. I have been thinking about moving them out into their own package or file once everything is finalized and verified as working. I'm just being a bit lazy and keeping it together to be able to work on it all in the same window for now, especially since things have been changing so quickly. 😅